php/libraries/NDB_Reliability.class.inc

```diff
@@ -151,13 +151,22 @@ class NDB_Reliability extends NDB_Form
      * Attempts to validate the form (using the defined rules) and
      * saves the validated data into the database
      *
-     * @return void
+     * @return bool
      * @access public
      */
     function save()
     {
-        if ($this->form->validate()) {
-            $success = $this->form->process(array(&$this, "_save"), false);
+        if (!$this->form->validate()) {
+            return false;
+        }
+        $success = $this->form->process(array(&$this, "_save"), false);
+        if (is_bool($success)) {
+            return $success;
+        } else {
+            throw new Exception(
```

Comment:
The parent class, `NDB_Form`, already has `save()`, this overrides it because these derived classes have their own extra checks. I wouldn't just pull code up a level if they're not semantically related or it'd actually make code harder to extend in future.

However, it has just been made clear that `process()` should always return a `bool` and that we do not want users to ever influence it to return anything else. With that decision in mind, maybe things will be different. I'll work on this pull request and see what my noggin' comes up with.